### ISSUE TITLE
Issue with parsing dates

### ISSUE THREAD

**Issue by nickhristov** [YugaByte/yugabyte-db#7]
Folks,

I am having an issue with parsing dates. Please look at this example:

```
cqlsh:location_data> SELECT * FROM location_data WHERE ld_id =  8a9b98ca-3fcb-4922-b735-599c03905886 AND device_created_on <= '2017-12-21 00:00:01.000000+0000';

 ld_id | device_created_on | latitude | longitude | total_distance | moving_status | speed_units | speed_value | status | status_key | status_id | total_halt_time | is_long_halt | summary_update | asset_summary_update | raw_data
-------+-------------------+----------+-----------+----------------+---------------+-------------+-------------+--------+------------+-----------+-----------------+--------------+----------------+----------------------+----------

(0 rows)
cqlsh:location_data> SELECT * FROM location_data WHERE ld_id =  8a9b98ca-3fcb-4922-b735-599c03905886 AND device_created_on <= '2017-12-21 00:00:01.000+00';

 ld_id                                | device_created_on               | latitude   | longitude | total_distance | moving_status | speed_units | speed_value | status    | status_key | status_id | total_halt_time | is_long_halt | summary_update | asset_summary_update | raw_data
--------------------------------------+---------------------------------+------------+-----------+----------------+---------------+-------------+-------------+-----------+------------+-----------+-----------------+--------------+----------------+----------------------+-------------------------------
 8a9b98ca-3fcb-4922-b735-599c03905886 | 2017-12-21 00:00:01.000000+0000 | -119.00867 |  36.07972 |        0.02488 |             1 |        KM/H |          76 | At pickup |       2104 |    100164 |               0 |        False |          False |                False | { ..... yada yada yada .......}
 8a9b98ca-3fcb-4922-b735-599c03905886 | 2017-12-21 00:00:00.000000+0000 | -119.00866 |  36.07936 |              0 |             1 |        null |           0 | At pickup |       2104 |    100164 |               0 |        False |           True |                False |  { ..... yada yada yada .......}

(2 rows)
```

**Comment by m-iancu** [comment]
@nickhristov Thanks for reporting this. We only support 3 fractional digits in the input syntax (milliseconds precision), so we should return an invalid format error for the input in your first example (`2017-12-21 00:00:01.000000+0000`). 

Instead, it looks like the extra digits confuse the timestamp parser and cause the unexpected result. I will look into why we don't catch that.
Meanwhile, a simple workaround is to replace the first entry with `2017-12-21 00:00:01.000+0000`. 

Note: Here are the timestamp formats we currently support: https://docs.yugabyte.com/api/cql/type_timestamp/

**Comment by nickhristov** [comment]
Yup, this does not seem to be a problem when we use data bindings. And yes, the expectation that I would have is that it would fail in the parse stage of the query processing.

**Comment by m-iancu** [comment]
closed by 679b57b28c49d1bec9f4e57707eef48092677d07.
